src/UserNotes/UserNote.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace phpweb\UserNotes;
+
+class UserNote
+{
+    /** @var string $id */
+    public $id;
+    /** @var string $sect */
+    public $sect;
+    /** @var string $rate */
+    public $rate;
+    /** @var string $ts */
+    public $ts;
+    /** @var string $user */
+    public $user;
+    /** @var string $text */
+    public $text;
+    /** @var int $upvotes */
+    public $upvotes;
+    /** @var int $downvotes */
+    public $downvotes;
```

Comment:
How do you feel about marking these as `private` and adding accessors instead?

```suggestion
-    /** @var string $id */
-    public $id;
-    /** @var string $sect */
-    public $sect;
-    /** @var string $rate */
-    public $rate;
-    /** @var string $ts */
-    public $ts;
-    /** @var string $user */
-    public $user;
-    /** @var string $text */
-    public $text;
-    /** @var int $upvotes */
-    public $upvotes;
-    /** @var int $downvotes */
-    public $downvotes;
+    private $id;
+    private $sect;
+    private $rate;
+    private $ts;
+    private $user;
+    private $text;
+    private $upvotes;
+    private $downvotes;
```

Reply:
I'd rather go with `readonly` (as annotation for now), if the properties are actually readonly.

Reply:
@cmb69 

I think they are mutated currently.

Reply:
I experimented with few different approaches. In the end, I decided that it doesn't have to be immutable. With PHP 8.1, we might easily convert this into readonly and use constructor property promotion. It would make migration easier if we use public properties now. I'd keep it as it is now, optionally adding `readonly` annotation as cmb69 suggests.

Reply:
Thinking about it more, I decided to mark the class as `immutable` and `readonly`. I don't think any static analyser respects `readonly` annotation on class, but if we migrate to PHP 8.2, we will know to make the class as readonly.